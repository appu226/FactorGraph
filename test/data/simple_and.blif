.model DD
.inputs pi00 pi01 lo00
.outputs po00 li00 ans_qpi
.names pi00 pi01 lo00 li00
111 1
.names li00 lo00 po00
11 1
.names lo00 ans_qpi
1 1
.end
