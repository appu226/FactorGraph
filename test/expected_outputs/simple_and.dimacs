c ind 1 3 0
p cnf 6 12
2 0
-4 0
-5 -3 2 0
-5 3 4 0
5 3 -4 0
5 -3 -2 0
-6 -1 2 0
-6 1 -5 0
6 1 5 0
6 -1 -2 0
6 0
-6 0
