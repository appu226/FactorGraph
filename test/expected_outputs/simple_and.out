[DEBUG] parsing var pi00 as:
1--  1
[DEBUG] parsing var pi01 as:
-1-  1
[DEBUG] parsing var lo00 as:
--1  1
[DEBUG] creating var li00 as:
---1  1
[DEBUG] parsing circuit for li00 as:
111-  1
[DEBUG] Added 1 factors to partition number 1
[DEBUG] Printing factor: 
0--0  1
10-0  1
1100  1
1111  1
[INFO] Executing over approximating method FactorGraphApprox
[INFO] processing 1 partitions
[INFO] largestSupportSet: 30, largestBddSize: 1000000000, numConvergence: 1
[INFO] Initiating message passing
[INFO] Factor graph messages have converged in 3 iterations
[INFO] Ran 1 FactorGraph convergences
[INFO] Over approximating method FactorGraphApprox finished with 3 solutions.
[INFO] Executing under approximating method ExactAndAccumulate
[INFO] processing 1 partitions
[DEBUG] Conjunction of factors in partition
0--0  1
10-0  1
1100  1
1111  1
[DEBUG] Quantified result from partition
--00  1
--1-  1
[INFO] Under approximating method ExactAndAccumulate finished with 3 solutions.
[DEBUG] Writing diff to temp/simple_and.dimacs
[DEBUG] dumping cnf for upper limit func
--00  1
--1-  1
[DEBUG] dumping left branch 
----  1
[DEBUG]   adding 1 clause '4 0' for func one
[DEBUG] dumping right branch 
---0  1
[DEBUG]   already written
[DEBUG] adding 4 clauses on tseytin vars 3 1 4 -2 for bdd 
--00  1
--1-  1
[DEBUG] dumping cnf for func
--00  1
--1-  1
[DEBUG] already written
[DEBUG] cnfVar 1 represents bdd of index 2
--1-  1
[DEBUG] cnfVar 2 represents bdd of index 3
---1  1
[DEBUG] cnfVar 4 represents bdd 
----  1
[DEBUG] cnfVar 3 represents bdd 
--00  1
--1-  1
[INFO] SUCCESS simple_and.blif
